### PR TITLE
Database: update to EPSG v11.006

### PR DESCRIPTION
**Pull request by jjimenezshaw**

### CODE REVIEW

File: test/unit/test_operationfactory.cpp [hunk 1]
```diff
@@ -5871,8 +5882,10 @@ TEST(
         EXPECT_EQ(list[0]->nameStr(),
                   "Inverse of NAD83(CSRS)v6 to CGVD28 height (1) + "
                   "NAD83(CSRS)v6 to CGVD2013(CGG2013) height (1) "
-                  "using Ballpark geographic offset "
-                  "from NAD83(CSRS) to NAD83(CSRS)v6");
+                  "using Null geographic offset "
```
Comment (jjimenezshaw) [inline test/unit/test_operationfactory.cpp:5885]: Changed the test to pass after EPSG update. But it seems to be wrong.

**Comment by jjimenezshaw** [comment]
It includes the changes done for 11.005
I made some "atomic" commits to control those changes.
Once we like it, I can compact the commits.

File: scripts/build_db.py [hunk 2]
```diff
@@ -784,7 +784,7 @@ def fill_grid_transformation(proj_db_cursor):
         # 1126: Vertical change by geoid grid difference (NRCan)
         # WARNING: update Transformation::isGeographic3DToGravityRelatedHeight()
         # in src/iso19111/operation/singleoperation.cpp if adding new methods
-        elif method_code in (1071, 1080, 1081, 1083, 1084, 1085, 1088, 1089, 1090, 1091, 1092, 1093, 1094, 1095, 1096, 1097, 1098, 1100, 1101, 1103, 1105, 1110, 1112, 1113, 1114, 1115, 1118, 1122, 1124, 1126) and n_params == 2:
+        elif method_code in (1071, 1080, 1081, 1083, 1084, 1085, 1088, 1089, 1090, 1091, 1092, 1093, 1094, 1095, 1096, 1097, 1098, 1100, 1101, 1103, 1105, 1110, 1112, 1113, 1114, 1115, 1118, 1122, 1124, 1126, 1128, 1129) and n_params == 2:
```
Comment (rouault) [inline scripts/build_db.py:787]: As indicated in the above warning, Transformation::isGeographic3DToGravityRelatedHeight() in src/iso19111/operation/singleoperation.cpp needs to be udated

File: test/unit/test_operationfactory.cpp (hunk 1 above)
Comment (rouault) [reply test/unit/test_operationfactory.cpp:5885]: I believe this is correct. This is due to the "Canada - CGVD28" extent being extented to New Foundland, which adds a new transformation for it. The previous one using the ballpark geographic offset is then sorted in second position. But the test should be modified to ASSERT_GE(list.size(), 2U); and check list[1] instead. Checking the updated list[0] doesn't hurt either.

File: scripts/build_db.py (hunk 2 above)
Comment (jjimenezshaw) [reply scripts/build_db.py:787]: done

File: test/unit/test_operationfactory.cpp (hunk 1 above)
Comment (jjimenezshaw) [reply test/unit/test_operationfactory.cpp:5885]: done

**Comment by rouault** [comment]
@jjimenezshaw Looks good to me. I've also pushed commit c1becb63406b670e7d09d4d7e046032853816a2e to your branch to handle another method that was needed for
```
$ PROJ_DATA=data bin/projinfo -k operation EPSG:10557
PROJ string:
+proj=pipeline
  +step +proj=axisswap +order=2,1
  +step +proj=unitconvert +xy_in=deg +xy_out=rad
  +step +inv +proj=vgridshift +grids=dkmsl_2022.tif +multiplier=1
  +step +proj=axisswap +order=1,2,-3
  +step +proj=unitconvert +xy_in=rad +xy_out=deg
  +step +proj=axisswap +order=2,1

WKT2:2019 string:
COORDINATEOPERATION["ETRS89 to DKMSL(2022) depth (1)",
    VERSION["SDFI-Dnk 2022"],
    SOURCECRS[
        GEOGCRS["ETRS89",
            ENSEMBLE["European Terrestrial Reference System 1989 ensemble",
                MEMBER["European Terrestrial Reference Frame 1989"],
                MEMBER["European Terrestrial Reference Frame 1990"],
                MEMBER["European Terrestrial Reference Frame 1991"],
                MEMBER["European Terrestrial Reference Frame 1992"],
                MEMBER["European Terrestrial Reference Frame 1993"],
                MEMBER["European Terrestrial Reference Frame 1994"],
                MEMBER["European Terrestrial Reference Frame 1996"],
                MEMBER["European Terrestrial Reference Frame 1997"],
                MEMBER["European Terrestrial Reference Frame 2000"],
                MEMBER["European Terrestrial Reference Frame 2005"],
                MEMBER["European Terrestrial Reference Frame 2014"],
                MEMBER["European Terrestrial Reference Frame 2020"],
                ELLIPSOID["GRS 1980",6378137,298.257222101,
                    LENGTHUNIT["metre",1]],
                ENSEMBLEACCURACY[0.1]],
            PRIMEM["Greenwich",0,
                ANGLEUNIT["degree",0.0174532925199433]],
            CS[ellipsoidal,3],
                AXIS["geodetic latitude (Lat)",north,
                    ORDER[1],
                    ANGLEUNIT["degree",0.0174532925199433]],
                AXIS["geodetic longitude (Lon)",east,
                    ORDER[2],
                    ANGLEUNIT["degree",0.0174532925199433]],
                AXIS["ellipsoidal height (h)",up,
                    ORDER[3],
                    LENGTHUNIT["metre",1]],
            ID["EPSG",4937]]],
    TARGETCRS[
        VERTCRS["DKMSL(2022) depth",
            VDATUM["Denmark Mean Sea Level (2022)"],
            CS[vertical,1],
                AXIS["depth (D)",down,
                    LENGTHUNIT["metre",1]],
            ID["EPSG",10547]]],
    METHOD["Geographic3D to Depth (gtg)",
        ID["EPSG",1127]],
    PARAMETERFILE["Geoid (height correction) model file","dkmsl_2022.tif"],
    OPERATIONACCURACY[0.5],
    USAGE[
        SCOPE["Derivation of depths from GNSS observations."],
        AREA["4756 - Denmark - offshore"],
        BBOX[54.36,3.25,58.27,16.51]],
    ID["EPSG",10557],
    REMARK["For reversible alternative to this transformation see ETRS89 to ETRS89 + DKMSL(2022) depth (1) (code 10558). File format is Geodetic TIFF Grid (GTG), algorithmically equivalent to the GTX format."]]
```

@kbevers I guess you'll have follow-up actions to add the grids. By the way I find the comment of EPSG quite offensive "File format is Geodetic TIFF Grid (GTG), algorithmically equivalent to the GTX format." . GTG is a GTX on steroid !

**Review by rouault — APPROVED** [review]

**Comment by jjimenezshaw** [comment]
@rouault thanks
I will squash some of the commits later today or tomorrow. Do we want to make explicit the version 11.005 with its own commit, or should I squash it with 11.006 (and the fixes and corrections needed accordingly)?

**Comment by rouault** [comment]
> I will squash some of the commits later today or tomorrow

I could also just use the "squash & merge" github button. A single commit would do. For git bisect purposes, I tend to prefer commits that individually pass build+tests, and here I believe everything is almost tied up, except perhaps the "[do not create file builtin_authorities.sql](https://github.com/OSGeo/PROJ/pull/4085/commits/f4691a398c05f908d5259c16228257521afa6753) commit which is actually a bug fix and could be a separate leading commit

@kbevers do you want this in the 9.4 maintenance branch?

**Comment by jjimenezshaw** [comment]
> @kbevers do you want this in the 9.4 maintenance branch?

Usually we include EPSG db updates in maintenance branches, right? otherwise the user has to wait up to 6 months to get new stuff. (at least that was my impression).

**Comment by jjimenezshaw** [comment]
> here I believe everything is almost tied up, except perhaps the "[do not create file builtin_authorities.sql](https://github.com/OSGeo/PROJ/pull/4085/commits/f4691a398c05f908d5259c16228257521afa6753) commit which is actually a bug fix and could be a separate leading commit

Now it is two commits.

**Comment by rouault** [comment]
backport of 2nd commit to 9.4 in https://github.com/OSGeo/PROJ/pull/4086

**Pull request merged**